fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/DruidQuery.java

```diff
@@ -64,7 +64,7 @@
      */
     @JsonIgnore
     default DruidQuery<?> getInnermostQuery() {
-        return getInnerQuery().map(DruidQuery.class::cast).orElse(this);
+        return getInnerQuery().isPresent() ? getInnerQuery().get().getInnermostQuery() : this;
```

Comment:
Why not just do

```java
   return getInnerQuery.map(DruidQuery::getInnermostQuery).orElse(this);
```

Reply:
Null pointer exception if getInnerQuery returns null right?

Reply:
Ok you are right

Reply:
I'm getting a type conflict using this map method, might as well keep it what it is right now since both are basically equivalent.

Reply:
Apparently, IntelliJ is stumbling over the template types that neither Maven nor Eclim care about.

Stupid IntelliJ/complicated use of types that confuse our tools or something.